sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -483,41 +532,115 @@ public void validate(PipelineOptions options) {
 
     @Override
     public PCollection<TableRow> expand(PBegin input) {
-      final String stepUuid = BigQueryHelpers.randomUUIDString();
-      BoundedSource<TableRow> source;
+      Pipeline p = input.getPipeline();
+      final String staticJobUuid = BigQueryHelpers.randomUUIDString();
+
+      final PCollectionView<String> jobIdTokenView;
+      PCollection<String> jobIdTokenCollection = null;
+      if (!getUseNewSource()) {
+        // Create a singleton job ID token at construction time.
+        jobIdTokenView = p
+            .apply("TriggerIdCreation", Create.of(staticJobUuid))
+            .apply("ViewId", View.<String>asSingleton());
+      } else {
+        // Create a singleton job ID token at execution time.
+        jobIdTokenCollection = p
+            .apply("TriggerIdCreation", Create.of("ignored"))
+            .apply("CreateJobId", MapElements.via(
+                new SimpleFunction<String, String>() {
+                  @Override
+                  public String apply(String input) {
+                    return BigQueryHelpers.randomUUIDString();
+                  }
+                }));
+        jobIdTokenView = jobIdTokenCollection
+            .apply("ViewId", View.<String>asSingleton());
+      }
 
-      if (getQuery() != null
-          && (!getQuery().isAccessible() || !Strings.isNullOrEmpty(getQuery().get()))) {
-        source =
-            BigQueryQuerySource.create(
-                stepUuid,
-                getQuery(),
-                getFlattenResults(),
-                getUseLegacySql(),
-                getBigQueryServices());
+      PCollection<TableRow> rows;
+      if (!getUseNewSource()) {
+        // Apply the traditional Source model.
+        rows = p.apply(
+            org.apache.beam.sdk.io.Read.from(applySourceTransform(staticJobUuid)))
+            .setCoder(getDefaultOutputCoder());
       } else {
-        source =
-            BigQueryTableSource.create(
-                stepUuid,
-                getTableProvider(),
-                getBigQueryServices());
+        // Apply the DoFn version.
+        final TupleTag<String> filesTag =
+            new TupleTag<String>(){};
```

Comment:
Could you point me to where I want setCoder?  MultiOutput doesn't have coder methods, and the docs appear to advocate for "{}": https://beam.apache.org/documentation/programming-guide/#transforms-sideio